pkg/azuredisk/controllerserver.go

```diff
@@ -847,58 +760,37 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 	}
 
 	snapshotName = azureutils.CreateValidDiskName(snapshotName)
-
-	var customTags string
-	// set incremental snapshot as true by default
-	incremental := true
-	var resourceGroup string
 	var err error
 	localCloud := d.cloud
-	location := d.cloud.Location
-
-	parameters := req.GetParameters()
-	for k, v := range parameters {
-		switch strings.ToLower(k) {
-		case consts.TagsField:
-			customTags = v
-		case consts.IncrementalField:
-			if v == "false" {
-				incremental = false
-			}
-		case consts.ResourceGroupField:
-			resourceGroup = v
-		case consts.LocationField:
-			location = v
-		case consts.UserAgentField:
-			newUserAgent := v
-			localCloud, err = azureutils.GetCloudProvider(d.kubeconfig, d.cloudConfigSecretName, d.cloudConfigSecretNamespace, newUserAgent, d.allowEmptyCloudConfig)
-			if err != nil {
-				return nil, fmt.Errorf("create cloud with UserAgent(%s) failed with: (%s)", newUserAgent, err)
-
-			}
-		default:
-			return nil, fmt.Errorf("AzureDisk - invalid option %s in VolumeSnapshotClass", k)
+	params := req.GetParameters()
+	diskParams, err := azureutils.ParseDiskParameters(params)
```

Comment:
could you only make changes to volume cap verification? `ParseDiskParameters` func implementation is mainly for CreateVolume, and it does not apply to CreateSnapshot and ExpandVolume since different CSI interface allows different parameters